homeassistant/helpers/template.py

```diff
@@ -571,7 +574,7 @@ def async_render_with_possible_json_value(
             return self.template
 
         if self._compiled is None:
-            self._ensure_compiled()
+            self._compiled = self._ensure_compiled()
```

Comment:
I think this is nicer than adding an `assert` after `ensure_compiled`.

Reply:
It sounds like `_ensure_compiled` should be changed to not have the side effect of assigning `self._compiled` then...? (And probably renamed, at that point.)

Reply:
You're right - it was awkard.

I've decided to align the code style with other similar call inside the file.